### ISSUE TITLE
Report a warning if ghc-mod was compiled with a different version of GHC

### ISSUE THREAD

**Issue by orome** [carymrobbins/intellij-haskforce#219]
After updating GHC to 7.10.3 (using Homebrew), HaskForce stops functioning. For example I get
```
cannot satisfy -package-id MissingH-1.3.0.1-b2ed9a940ca5230e3bce244404ce0af7
```
and the "External Libraries" still link to the 7.10.2 folder (even though that's not where new installations are going.)

In this case I have MissingH 1.3.0.1 in moth the 7.10.2 folder and the 7.10.3 folder. I'ver restarted and invalidated coaches, but to no avail.

What's the procedure for getting HaskForce working (and using the correct lib folder) following an update of GHC?

**Comment by orome** [comment]
To get things working, it seems I just need to (1) rebuild by project and (2) reinstall and reset all the tools (e.g. `ghc-mod`) — but the libraries are still listed as wrong: #220.

**Comment by carymrobbins** [comment]
Maybe we should check that the configured ghc-mod was built with the configured GHC version.  If they don't match, report a warning.

**Comment by orome** [comment]
@carymrobbins — Yes, as above, if I reset and reinstall a new `ghc-mod`, it says it's built by the new GHC.

**Comment by carymrobbins** [comment]
This goes well with #229.  I assume, @orome, that you are using Cabal; however, we should account for both scenarios (cabal and stack).

We can see the version of ghc used to compile ghc-mod by doing the following - 

```
$ ghc-mod --version
ghc-mod version 5.5.0.0 compiled by GHC 7.10.3
```

With a cabal project, we can just check our configured ghc, ask it for its version, and confirm that it matches the version our configured ghc-mod was compiled with.  If we are using stack, we can do the same except check the ghc version using the following -

```
$ stack exec -- ghc --version    
The Glorious Glasgow Haskell Compilation System, version 7.10.3
```

If the versions don't match, we should report this warning to the user.  It's probably useful to run this check when first opening a project, and then again if the ghc-mod path changes.

**Comment by orome** [comment]
If I run
```
$ stack exec -- ghc --version
```
I get
```
Run from outside a project, using implicit global project config
Using resolver: lts-3.7 from implicit global project's config file: /Users/Orome/.stack/global/stack.yaml
Caching build plan
Populated index cache.    
Compiler version mismatched, found ghc-7.10.3 (x86_64), but expected minor version match with ghc-7.10.2 (x86_64) (based on resolver setting in /Users/Orome/.stack/global/stack.yaml).
Try running "stack setup" to install the correct GHC into /Users/Orome/.stack/programs/x86_64-osx/
```
What's going on there?

**Comment by carymrobbins** [comment]
Actually, it's probably better to do `stack ghc -- --version` to invoke `ghc` instead of the way I was doing it, `stack exec -- ghc --version`.  I've updated my original comment to use the proper command instead.

@orome - I assume that your project is build with cabal instead of stack.  Either way, try running that command that I suggested instead.  You'll probably get the same warning, so then you may want to do `stack setup` if you plan on using stack as your builder.

**Comment by orome** [comment]
I use Cabal but may eventually switch to Stack. Should I run `stack setup` now in anticipation just to be tidy? Are there side effects I should be concerned about?

**Comment by carymrobbins** [comment]
I don't think there are any unintended side-effects.  Stack is particularly hygienic, so you don't have to worry much about its binaries effecting things you already have installed.

**Comment by orome** [comment]
I went ahead and ran `stack setup` which seems to have worked without side effects. Before doing so though it was important to change the `resolver` in `~/.stack/global/stack.yaml` to something that uses the latest `ghc`.

**Comment by carymrobbins** [comment]
Good point, I should have mentioned that as well.  I usually just go to https://www.stackage.org/snapshots and reach for the latest LTS version.

**Comment by carymrobbins** [comment]
This has been resolved in the latest release, **v0.3-beta.27**.
